(Samples) Transfer in (#231)
OfficeDev/office-scripts-docs: docs/resources/samples/add-excel-comments.md

---
title: 'Add comments in Excel'
description: 'Learn how to use Office Scripts to add comments in a worksheet.'
ms.date: 03/29/2021
localization_priority: Normal
---

# Add comments in Excel

This sample shows how to add comments to a cell including [@mentioning](https://support.microsoft.com/office/90701709-5dc1-41c7-aa48-b01d4a46e8c7) a colleague.

## Example scenario

* The team lead maintains the shift schedule. The team lead assigns an employee ID to the shift record.

> * The team lead wishes to notify the employee. By adding a comment that @mentions the employee, the employee is emailed with a custom message from the sheet.

Reviewer: ```suggestion
* The team lead wishes to notify the employee. By adding a comment that @mentions the employee, the employee is emailed with a custom message from the worksheet.
```

> * The employee can then address and respond to the comment.

Reviewer: Should we mention this isn't in the script?
```suggestion
* The employee can view the workbook and respond to the comment at their convenience.
```

Author: I don't think so. The scenario seems to discuss a usage flow then the solution section discuss what the script actually does.

## Solution

> 1. The Office Script extracts employee information from the employee sheet.

Reviewer: Let's make this less formal.
```suggestion
1. The script extracts employee information from the employee worksheet.
```

> 1. The script then cross-references the shift record and adds a comment to the appropriate cell by including the relevant employee email in the team lead's comment.

Reviewer: Can we break this into two sentences (and two steps)? Like:
1. The script then cross-references the shift record.
1. It adds a comment to the appropriate cell that includes the relevant employee email.

> 1. It also removes any existing comment on the cell before adding a new comment.

Reviewer: ```suggestion
1. Existing comments in the cell are removed before adding the new comment.
```

> ## Office Scripts sample code: Add comments

Reviewer 2: Elizabeth, you can remove "Office Scripts" from the H2's so that it reads:
 ## Sample code: Add comment or ## Sample code: Add comments in Office Scripts. Ditto training video. That way the repetition is not at the front of the line. If you do remove "Office Scripts" from the H2 ##, I don't think SEO will be affected that much since "Office Scripts" is in the metadata.

Download the file <a href="excel-comments.xlsx">excel-comments.xlsx</a> used in this sample and try it out yourself!

```TypeScript

> function main(workbook: ExcelScript.Workbook) {

Reviewer: In addition to the comments, let's create a another issue to standardize all the samples (like we did a couple years ago for office-js-snippets).

Author: Tracking with feature 4924924

---
title: 'Add comments in Excel'
description: 'Learn how to use Office Scripts to add comments in a worksheet.'
ms.date: 03/29/2021
localization_priority: Normal
---

# Add comments in Excel

This sample shows how to add comments to a cell including [@mentioning](https://support.microsoft.com/office/90701709-5dc1-41c7-aa48-b01d4a46e8c7) a colleague.

## Example scenario

* The team lead maintains the shift schedule. The team lead assigns an employee ID to the shift record.
* The team lead wishes to notify the employee. By adding a comment that @mentions the employee, the employee is emailed with a custom message from the sheet.
* The employee can then address and respond to the comment.

## Solution

1. The Office Script extracts employee information from the employee sheet.
1. The script then cross references the shift record and adds a comment to the appropriate cell by including the relevant employee email in the team lead's comment.
1. It also removes any existing comment on the cell before adding a new comment.

## Sample Excel file

Download the file <a href="excel-comments.xlsx">excel-comments.xlsx</a> used in this sample and try it out yourself!

> ## Office Script

Reviewer 2: ```suggestion
## Office Script sample code: Add comments
```

> ```TypeScript

Reviewer 2: Add code comments

Author: Yeah, I had received this feedback before. We're tracking with this item: 4958338
